### ISSUE TITLE
Call list cases should time out after 8 hours

### ISSUE THREAD

**Issue by colinxfleming** [colinxfleming/dcaf_case_management#118]
From discussions with @lwaldsc about call list depopulation behavior: 

* A case manager should begin their night with an empty call list 
* Any calls from the previous evening should time out after a certain amount of time 
* We can't do this based on user session because someone might have to restart and that would wipe away their call list

**Comment by colinxfleming** [comment]
This came up during the pilot, so I'm promoting it to the front for release. I also want to spec out exactly how it should behave before we start programming on it. Tagging in @lwaldsc, @eheintzelman , @SabrinaRitacco   and @NerdyGirl537  about best way to implement here. 

Here are two system behavior options to get us started:
* Can we safely say that every day at like 4am, we should clear out everyone's call lists? (This is probably the path of least resistance.)
* We could also set it up so that a call on the call list timed out after 12 hours; eg. a patient added to the call list at 7pm would be removed at 7am.

**Comment by colinxfleming** [comment]
topping for @lwaldsc @NerdyGirl537 @SabrinaRitacco @eheintzelman . I'm inclined to say that at 4am we run a scheduled system job that clears out every single call list no matter what - what do you all think?

**Comment by mebates** [comment]
Makes sense to me.

**Comment by NerdyGirl537** [comment]
What about looking at status?  If I haven't reached a PT, I want to keep them in the call list.  If they disappear, I might not remember who I didn't reach.  For the rest, 4 AM seems reasonable.

**Comment by colinxfleming** [comment]
I think that's a useful insight from @NerdyGirl537 . I think we can turn this into a real task now, as follows:

* Add a `clear_call_list` method to the user model that iterates through every call in `user.pregnancies` and removes it if the patient hasn't been reached yet
* Create a rake task that calls the 'clear_call_list' on every user (see https://devcenter.heroku.com/articles/scheduler#defining-tasks for an example of how we'll use this)
* Flag for @colinxfleming when this is done and we'll make it a scheduled task via heroku's scheduler

**Comment by mebates** [comment]
Hey guys - this issue is a bit confusing. Can you summarize where we landed on the rules about clearing calls (in plain language)? Need the info for reporting discussion... Thanks!

**Comment by colinxfleming** [comment]
The logic here shook out as follows:

We remove you from a user's call list at 4am nightly if all of the following are true:
* you called the patient
* and reached them
* within the past 8 hours

**Comment by mebates** [comment]
So what came up tonight was the many patients who are 'dormant' for days, weeks, or forever. We don't want to encourage CMs to keep calling PTs who don't give DCAF a call back, and leaving them on the list for the whole shift might do that. And what if you come back a month later and they're still there? Or 9 months?

**Comment by NerdyGirl537** [comment]
Okay, I'm working through some different thoughts here.  Perhaps, we need something as part of the handoff/transition between shifts that will clear the CMs call log.  Since we're considering some sort of report for the CM to share, perhaps that can also trigger clearing the call list?
